compiler/ast/typesrenderer.nim

```diff
@@ -264,18 +264,24 @@ proc typeToString*(typ: PType, prefer: TPreferedDesc = preferName): string =
     of tyBuiltInTypeClass:
       result = case t.base.kind
         of tyVar: "var"
+        of tyLent: "lent"
```

Comment:
Since `var` is allowed as a type class, my thinking was that `lent` should be allowed too, as it may also be used as the type of a field, local, or result variable.

`sink` only annotates a parameter as having full ownership of the value, and the type cannot appear outside of parameter definitions. Outside of destruction injection and move analysis, it is as if `sink` (`tySink`) doesn't exist, whereas this is not the case for `lent`.

---

At present, both `var` and `lent` have the problem that `x is var` and `x is lent` checks don't work due to `semExprWithType` always inserting a hidden dereference, but this is fixable/changeable.